hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/JavaDefaultSerializers.java

```diff
@@ -220,6 +230,9 @@ public Object read(final ObjectDataInput in) throws IOException {
                 } else {
                     result = objectInputStream.readUnshared();
                 }
+                if (gzipEnabled) {
+                    gZipInputStream.pushBackUnconsumedBytes(in);
```

Comment:
do you have to pass the `in`? it's already passed in the constructor, isnt it?

Reply:
Ohh. right. I missed that.